### PR TITLE
Refactor LoadModification GroovyScript and EquipmentDeletion with generic design

### PR DESCRIPTION
**Pull request by ayolab**
Signed-off-by: Ayoub LABIDI <ayoub.labidi@protonmail.com>

### CODE REVIEW

**Review by laurgarn — COMMENTED** [review]
Code seems OK

File: src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java [hunk 1]
```diff
@@ -1338,424 +1193,6 @@ public void testCreateTwoWindingsTransformerInMixedTopology() throws Exception {
...
-
-        // delete load with error removing dangling switches, because the load connection node is not linked to any other node
-        mockMvc.perform(post(URI_NETWORK_MODIF).content(equipmentDeletionInfosJson).contentType(MediaType.APPLICATION_JSON))
-                .andExpect(status().isOk())
-                .andReturn();
-
-        var v5 = network.getVoltageLevel("v5");
-        assertNull(v5.getNodeBreakerView().getTerminal(2));
-    }
-
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java:1758]: I'm not sure that all the tests have been preserved

**Review by dbraquart — CHANGES_REQUESTED** [review]
tests: ok
code: a few changes to address

File: src/main/java/org/gridsuite/modification/server/modifications/GroovyScriptModification.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+/**
```
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/main/java/org/gridsuite/modification/server/modifications/GroovyScriptModification.java]: good job !

**Review by dbraquart — CHANGES_REQUESTED** [review]
Front-end tests : ok
review: ok, just a modifier to add please

File: src/main/java/org/gridsuite/modification/server/entities/GroovyScriptEntity.java [hunk 3]
```diff
@@ -22,27 +25,37 @@
...
+        assignAttributes(groovyScriptInfos);
+    }
+
+    @Override
+    public void update(@NonNull ModificationInfos modificationInfos) {
+        super.update(modificationInfos);
+        assignAttributes((GroovyScriptInfos) modificationInfos);
+    }
+
+    void assignAttributes(GroovyScriptInfos modificationInfos) {
```
[the remark this answers is not in the document]
Comment (ayolab) [reply src/main/java/org/gridsuite/modification/server/entities/GroovyScriptEntity.java]: Done

**Review by dbraquart — APPROVED** [review]

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**